src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -118,5 +99,40 @@ public boolean isValidURL(String url) {
       return false;
     }
   }
+
+  public String displayMedia(String regexURL, String userInput) {
+    String replacement;
+    String textWithMediaReplaced = userInput;
+
+    Pattern pattern = Pattern.compile(regexURL);
+    Matcher matcher = pattern.matcher(userInput);
+
+    while (matcher.find()) {
+      String mediaURL = matcher.group(1);
+      System.out.println("A media url found: " + mediaURL);
+
+      if (isValidURL(mediaURL)) {
+        if (mediaURL.endsWith(".mp3")) {
+          replacement = "<audio controls src=" + mediaURL + " />";
+          textWithMediaReplaced = textWithMediaReplaced.replace(mediaURL, replacement);
+          System.out.println("URL changed with audio tag: " + textWithMediaReplaced);
+        } else if (mediaURL.endsWith(".mp4")) {
+          replacement = "<video controls src=" + mediaURL + " />";
+          textWithMediaReplaced = textWithMediaReplaced.replace(mediaURL, replacement);
+          System.out.println("URL changed with video tag: " + textWithMediaReplaced);
+        } else {
+          replacement = "<img src=" + mediaURL + " />";
```

Comment:
should we be explicit about which extensions are image tags?

It's assumed that if caller passed in anything that's not mp3/mp4 it would become an img tag, but that seems a bit error prone if we add support for a new type in the future

Reply:
True true, I agree could be problematic as the extensions grow! Originally I was thinking the regex would limit the url types found so could just specify the mp3/mp4 that were different and leave the rest as img haha. Btw do you have any suggestions to combine the 5 img extension types as I was thinking `||` for the `endsWith()` condition 5 times seems long and repetitive?

Reply:
the main issue i wanted to point out is that the caller is controlling which extensions to find, so it may not stay in sync with this file.

For the repeated list logic, i think it's not too bad. if you want to look into some APIs you can use java's streams api!

`List<String> imageExtensions = new ... // create a list of allowed extensions`
`imageExtensions.stream().anyMatch(ext -> mediaURL.endsWith(ext);)

this goes through the element in imageExtensions one by one, and returns true if any of them causes `mediaURL.endsWith(ext)` to return true.

`

Reply:
Ahhh okok, thanks! Will make the changes :)